synapse/lib/stormtypes.py

```diff
@@ -6397,10 +6408,21 @@ async def _methCronAdd(self, **kwargs):
                 'incvals': incval,
                 'creator': self.runt.user.iden
                 }
+
         iden = kwargs.get('iden')
         if iden:
             cdef['iden'] = iden
 
+        view = kwargs.get('view')
+        if not view:
+            todo = s_common.todo('getView', user=self.runt.user)
+            vdef = await self.runt.dyncall('cortex', todo)
+            if not vdef:
+                mesg = f'Could not get default view for user {self.runt.user.iden}'
+                raise s_exc.StormRuntimeError(mesg=mesg, kwargs=kwargs)
+            view = vdef.iden
+        cdef['view'] = view
+
         todo = s_common.todo('addCronJob', cdef)
         gatekeys = ((self.runt.user.iden, ('cron', 'add'), None),)
```

Comment:
Cron add permissions check should be updated to check against the view ( which is backward compat because it'll check the cortex as a universal check )